lib/bloc_record/selection.rb

```diff
@@ -204,6 +217,22 @@ def init_object_from_row(row)
 	end
 
 	def rows_to_array(rows)
-		rows.map { |row| new(Hash[columns.zip(row)]) }
+		rows.map { |row| init_object_from_row(row) }
```

Comment:
Nice reuse of another method here. I have always wondered why the curriculum writers decided not to do this.